JSQMessagesViewController/Model/JSQAudioMediaItem.m

```diff
@@ -0,0 +1,389 @@
+        [_playButton setImage:self.pauseButtonImage forState:UIControlStateSelected];
+        [_playButton addTarget:self action:@selector(onPlayButton:) forControlEvents:UIControlEventTouchUpInside];
+        [playView addSubview:_playButton];
+
+        // create a label to show the duration / elapsed time
+        NSString * durationString = [self timestampString:_audioPlayer.duration
+                                              forDuration:_audioPlayer.duration];
+        NSString * maxWidthString = [@"" stringByPaddingToLength:[durationString length] withString:@"0" startingAtIndex:0];
+        
+        // this is cheesy, but it centers the progress bar without extra space and
+        // without causing it to wiggle from side to side as the label text changes
+        CGSize labelSize = CGSizeMake(36,18);
+        if ([durationString length] < 4)
+            labelSize = CGSizeMake(18,18);
+        else if ([durationString length] < 5)
+            labelSize = CGSizeMake(24,18);
+        else if ([durationString length] < 6)
+            labelSize = CGSizeMake(30, 18);
+        CGRect labelFrame = CGRectMake(size.width - labelSize.width - self.controlInsets.right,
+                                       self.controlInsets.top, labelSize.width, labelSize.height);
+        _progressLabel = [[UILabel alloc] initWithFrame:labelFrame];
+        _progressLabel.textAlignment = NSTextAlignmentLeft;
+        _progressLabel.adjustsFontSizeToFitWidth = YES;
+        _progressLabel.textColor = self.tintColor;
+        _progressLabel.font = self.labelFont;
+        _progressLabel.text = maxWidthString;
+        
+        // sizeToFit adjusts the frame's height to the font
+        [_progressLabel sizeToFit];
+        labelFrame.origin.x = size.width - _progressLabel.frame.size.width - self.controlInsets.right;
+        labelFrame.origin.y =  ((size.height - _progressLabel.frame.size.height) / 2);
+        labelFrame.size.width = _progressLabel.frame.size.width;
+        labelFrame.size.height =  _progressLabel.frame.size.height;
+        _progressLabel.frame = labelFrame;
+        _progressLabel.text = durationString;
+        [playView addSubview:_progressLabel];
+
+        // create a progress bar
+        _progressView = [[UIProgressView alloc] initWithProgressViewStyle:UIProgressViewStyleDefault];
+        CGFloat xOffset = _playButton.frame.origin.x + _playButton.frame.size.width + self.controlPadding;
+        CGFloat width = labelFrame.origin.x - xOffset - self.controlPadding;
+        _progressView.frame = CGRectMake(xOffset, (size.height - _progressView.frame.size.height) / 2,
+                                          width, _progressView.frame.size.height);
+        _progressView.tintColor = self.tintColor;
+        [playView addSubview:_progressView];
+
+        [JSQMessagesMediaViewBubbleImageMasker applyBubbleImageMaskToMediaView:playView isOutgoing:self.appliesMediaViewMaskAsOutgoing];
+        self.cachedMediaView = playView;
```

Comment:
Let's pull all of this view construction out into a new class, `JSQAudioMediaView`

this should have an `applyConfiguration:` method to do with the config class described above.